repotracker/repotracker.go

```diff
@@ -256,94 +230,80 @@ func (repoTracker *RepoTracker) StoreRevisions(ctx context.Context, revisions []
-			// won't take down the worker.
-			panic(err)
-		}
+		var versionErrs *VersionErrors
 		project, err := repoTracker.GetProjectConfig(ctx, revision)
 		if err != nil {
-			projectError, isProjectError := err.(projectConfigError)
-			if isProjectError {
-				if len(projectError.Warnings) > 0 {
-					// Store the warnings and keep going. If we don't have
-					// any true errors, the version will still be created.
-					v.Warnings = projectError.Warnings
+			// this is an error that implies the file is invalid - create a version and store the error
+			projErr, isProjErr := err.(projectConfigError)
+			if isProjErr {
+				versionErrs = &VersionErrors{
+					Warnings: projErr.Warnings,
+					Errors:   projErr.Errors,
 				}
-				if len(projectError.Errors) > 0 {
-					// Store just the stub version with the project errors
-					v.Errors = projectError.Errors
-					if err = v.Insert(); err != nil {
-						grip.Error(message.WrapError(err, message.Fields{
-							"message":  "failed storing stub version in project",
+				if len(versionErrs.Errors) > 0 {
+					stubVersion, dbErr := shellVersionFromRevision(ref, revisions[i])
+					if dbErr != nil {
+						grip.Error(message.WrapError(dbErr, message.Fields{
+							"message":  "error creating shell version",
 							"project":  ref.Identifier,
 							"revision": revision,
-							"runner":   RunnerName,
-							"version":  v.Id,
 						}))
-						return nil, err
 					}
-					newestVersion = v
+					stubVersion.Errors = versionErrs.Errors
+					stubVersion.Warnings = versionErrs.Warnings
+					err = stubVersion.Insert()
+					grip.Error(message.WrapError(err, message.Fields{
+						"message":  "error inserting shell version",
+						"project":  ref.Identifier,
+						"revision": revision,
+					}))
+					newestVersion = stubVersion
 					continue
 				}
 			} else {
-				// Fatal error - don't store the stub
 				grip.Error(message.WrapError(err, message.Fields{
-					"message":  "could not store project stub",
+					"message":  "error getting project config",
 					"project":  ref.Identifier,
 					"revision": revision,
-					"runner":   RunnerName,
-					"version":  v.Id,
 				}))
 				return nil, err
 			}
 		}
 
-		// We have a config, so turn it into a usable yaml string to store with the version doc
-		projectYamlBytes, err := yaml.Marshal(project)
-		if err != nil {
-			return nil, errors.Wrap(err, "Error marshaling config")
-		}
-		v.Config = string(projectYamlBytes)
-
 		// "Ignore" a version if all changes are to ignored files
+		var ignore bool
 		if len(project.Ignore) > 0 {
 			var filenames []string
 			filenames, err = repoTracker.GetChangedFiles(ctx, revision)
 			if err != nil {
-				return nil, errors.Wrap(err, "error checking GitHub for ignored files")
+				grip.Error(message.WrapError(err, message.Fields{
+					"message":  "error checking GitHub for ignored files",
+					"project":  ref.Identifier,
+					"revision": revision,
+				}))
+				continue
```

Comment:
The philosophy should be continue-on-error semantics so that errors creating 1 version should not block creation of others unless the error indicates a problem with the db